src/symbols/analyzer.rs

```diff
@@ -455,6 +593,138 @@ impl Symbol {
+    match &self {
+      SymbolMemberDecl::AutoAccessor(n) => {
+        SymbolMemberNodeRef::AutoAccessor(n.value())
+      }
+      SymbolMemberDecl::ClassMethod(n) => {
+        SymbolMemberNodeRef::ClassMethod(n.value())
+      }
+      SymbolMemberDecl::ClassProp(n) => {
+        SymbolMemberNodeRef::ClassProp(n.value())
+      }
+      SymbolMemberDecl::Constructor(n) => {
+        SymbolMemberNodeRef::Constructor(n.value())
+      }
+      SymbolMemberDecl::TsIndexSignature(n) => {
+        SymbolMemberNodeRef::TsIndexSignature(n.value())
+      }
+      SymbolMemberDecl::TsCallSignatureDecl(n) => {
+        SymbolMemberNodeRef::TsCallSignatureDecl(n.value())
+      }
+      SymbolMemberDecl::TsConstructSignatureDecl(n) => {
+        SymbolMemberNodeRef::TsConstructSignatureDecl(n.value())
+      }
+      SymbolMemberDecl::TsPropertySignature(n) => {
+        SymbolMemberNodeRef::TsPropertySignature(n.value())
+      }
+      SymbolMemberDecl::TsGetterSignature(n) => {
+        SymbolMemberNodeRef::TsGetterSignature(n.value())
+      }
+      SymbolMemberDecl::TsSetterSignature(n) => {
+        SymbolMemberNodeRef::TsSetterSignature(n.value())
+      }
+      SymbolMemberDecl::TsMethodSignature(n) => {
+        SymbolMemberNodeRef::TsMethodSignature(n.value())
+      }
+    }
+  }
+
+  /// Range of the symbol member.
+  pub fn range(&self) -> SourceRange {
+    match self.node() {
+      SymbolMemberNodeRef::AutoAccessor(n) => n.range(),
+      SymbolMemberNodeRef::ClassMethod(n) => n.range(),
+      SymbolMemberNodeRef::ClassProp(n) => n.range(),
+      SymbolMemberNodeRef::Constructor(n) => n.range(),
+      SymbolMemberNodeRef::TsIndexSignature(n) => n.range(),
+      SymbolMemberNodeRef::TsCallSignatureDecl(n) => n.range(),
+      SymbolMemberNodeRef::TsConstructSignatureDecl(n) => n.range(),
+      SymbolMemberNodeRef::TsPropertySignature(n) => n.range(),
+      SymbolMemberNodeRef::TsGetterSignature(n) => n.range(),
+      SymbolMemberNodeRef::TsSetterSignature(n) => n.range(),
+      SymbolMemberNodeRef::TsMethodSignature(n) => n.range(),
+    }
+  }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum SymbolMemberNodeRef<'a> {
+  AutoAccessor(&'a AutoAccessor),
+  ClassMethod(&'a ClassMethod),
+  ClassProp(&'a ClassProp),
+  Constructor(&'a Constructor),
+  TsIndexSignature(&'a TsIndexSignature),
+  TsCallSignatureDecl(&'a TsCallSignatureDecl),
+  TsConstructSignatureDecl(&'a TsConstructSignatureDecl),
+  TsPropertySignature(&'a TsPropertySignature),
+  TsGetterSignature(&'a TsGetterSignature),
+  TsSetterSignature(&'a TsSetterSignature),
+  TsMethodSignature(&'a TsMethodSignature),
+}
+
+#[derive(Default, Copy, Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub struct SymbolMemberId(u32);
+
+impl std::fmt::Debug for SymbolMemberId {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    // for less verbose debugging
+    write!(f, "{}", self.0)
+  }
+}
+
+#[derive(Debug, Clone)]
+pub struct SymbolMember {
```

Comment:
`SymbolMember` could probably be generalized into `Symbol`, but it's a little bit challenging with the borrow checker in Rust. I will open a separate PR for that.

Reply:
This PR is actually superceeded by https://github.com/denoland/deno_graph/pull/322 , but this one needs to merge first because it's starting to get very complicated with all these branches I have.